### PR TITLE
Add the ability to overide the DO API address.

### PR DESCRIPTION
**Pull request by cagedmantis**

### CODE REVIEW

File: do/cloud.go [hunk 1]
```diff
@@ -63,11 +72,14 @@ func newCloud(config io.Reader) (cloudprovider.Interface, error) {
 	}
 
 	oauthClient := oauth2.NewClient(oauth2.NoContext, tokenSource)
-	doClient := godo.NewClient(oauthClient)
+	doClient, err := godo.New(oauthClient, opts...)
+	if err != nil {
+		return nil, errors.New("failed to create godo client")
```
Comment (andrewsykim) [inline do/cloud.go]: `fmt.Errorf()` to include the error in the return value

Comment (cagedmantis) [reply do/cloud.go]: Sure. Would you like me to do the same for the dropletRegion call?

Comment (andrewsykim) [reply do/cloud.go]: If you don't mind :)

**Comment by andrewsykim** [comment]
lgtm, thanks for the clean up!

**Pull request merged**